widget/button.go

```diff
@@ -13,7 +13,190 @@ import (
 	"fyne.io/fyne/theme"
 )
 
-const buttonTapDuration = 250
+// ButtonAlign represents the horizontal alignment of a button.
+type ButtonAlign int
+
+// ButtonIconPlacement represents the ordering of icon & text within a button.
+type ButtonIconPlacement int
+
+// ButtonImportance represents how prominent the button should appear
+type ButtonImportance int
+
+// ButtonStyle determines the behaviour and rendering of a button.
+type ButtonStyle int
+
+const (
+	// ButtonAlignCenter aligns the icon and the text centrally.
+	ButtonAlignCenter ButtonAlign = iota
+	// ButtonAlignLeading aligns the icon and the text with the leading edge.
+	ButtonAlignLeading
+	// ButtonAlignTrailing aligns the icon and the text with the trailing edge.
+	ButtonAlignTrailing
+)
+
+const (
+	// ButtonIconLeadingText aligns the icon on the leading edge of the text.
+	ButtonIconLeadingText ButtonIconPlacement = iota
+	// ButtonIconTrailingText aligns the icon on the trailing edge of the text.
+	ButtonIconTrailingText
+)
+
+const (
+	// MediumImportance applies a standard appearance.
+	MediumImportance ButtonImportance = iota
+	// HighImportance applies a prominent appearance.
+	HighImportance
+	// LowImportance applies a subtle appearance.
+	LowImportance
+)
+
+const (
+	// DefaultButton is the standard button style.
+	// Deprecated: use Importance = MediumImportance instead.
+	DefaultButton ButtonStyle = iota
+	// PrimaryButton that should be more prominent to the user.
+	// Deprecated: use Importance = HighImportance instead.
+	PrimaryButton
+
+	buttonTapDuration = 250
+)
+
+// Button widget has a text label and triggers an event func when clicked
+type Button struct {
+	DisableableWidget
+	Text string
+	// Deprecated, use Importance instead, where HighImportance matches PrimaryButton
+	Style ButtonStyle
+	Icon  fyne.Resource
+	// Specify how prominent the button should be, High will highlight the button and Low will remove some decoration.
+	Importance    ButtonImportance
+	disabledIcon  fyne.Resource
```

Comment:
nit: move down to other unexported fields `hovered, tapped bool`

Reply:
As discussed before the field order has not been specified in the order chat.
Before asking for it to be moved we should agree on the order.